src/linear_algebra/orientation.lean

```diff
@@ -0,0 +1,379 @@
+/-
+Copyright (c) 2021 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Myers
+-/
+import linear_algebra.determinant
+
+/-!
+# Orientations of modules and rays in modules
+
+This file defines rays in modules and orientations of modules.
+
+## Main definitions
+
+* `module.ray` is a type for the equivalence class of nonzero vectors in a module with some
+common positive multiple.
+
+* `orientation` is a type synonym for `module.ray` for the case where the module is that of
+alternating maps from a module to its underlying ring.  An orientation may be associated with an
+alternating map or with a basis.
+
+* `module.oriented` is a type class for a choice of orientation of a module that is considered
+the positive orientation.
+
+## Implementation notes
+
+`orientation` is defined for an arbitrary index type, but the main intended use case is when
+that index type is a `fintype` and there exists a basis of the same cardinality.
+
+## References
+
+* https://en.wikipedia.org/wiki/Orientation_(vector_space)
+
+-/
+
+noncomputable theory
+
+section ordered_comm_semiring
+
+variables (R : Type*) [ordered_comm_semiring R]
+variables {M : Type*} [add_comm_monoid M] [module R M]
+variables (ι : Type*) [decidable_eq ι]
+
+/-- Two vectors are in the same ray if some positive multiples of them are equal (in the typical
+case over a field, this means each is a positive multiple of the other).  Over a field, this
+is equivalent to `mul_action.orbit_rel`. -/
```

Comment:
Would you mind adding a proof of this statement about `field`s?

Reply:
I've added `same_ray_iff_orbit_rel`.